### PR TITLE
MAE-331: Add Linter, Test Actions

### PR DESCRIPTION
**Pull request by erawat**
### Overview

This PR adds

- Github Actions
  - Linters action to run phpcs linters again the php codes that have been added or updated when PR is created.
  - Test action to run phpunit tests to run tests against the repository when PR is created. 
- Github Pull Request Template
- PHPCS Ruleset 

This PR also changes the phpunit bootstrap level from classloader to settings, so we could fabricate the CiviCRM settings on unit tests.

### CODE REVIEW

File: .github/tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    steps:
+
+      - name: Config mysql database as per CiviCRM requirement
+        run: echo "SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));" | mysql -u root --password=root --host=mysql
+
+      - name: Config amp
+        run : amp config:set --mysql_dsn=mysql://root:root@mysql:3306
+
+      - name: Build Drupal site
+        run: civibuild create drupal-clean --civi-ver 5.24.6 --web-root $GITHUB_WORKSPACE/site
```
Comment (omarabuhussein) [inline .github/tests.yml:32]: I think we had a discussion about this in another PR but I kinda forgot, but we use a patched civicrm version and not the core one

File: .github/tests.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          repo: compucorp/civicrm-core
+          version: 5.24.6
+          path: site/web/sites/all/modules/civicrm
+
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.manualdirectdebit
```
Comment (omarabuhussein) [inline .github/tests.yml:44]: what this exactly do ?

Comment (erawat) [reply .github/tests.yml]: @omarabuhussein  good caught. updated.

File: .github/tests.yml (hunk 1 above)
Comment (erawat) [reply .github/tests.yml:32]: @omarabuhussein  yes, we have patched action to apply our patched.  see https://github.com/compucorp/uk.co.compucorp.membershipextras/pull/274/files#diff-ba72dd894f9b61a2c721a74670e376ccR34

Comment (omarabuhussein) [reply .github/tests.yml:32]: oh right

**Pull request merged**